src/printer.cpp

```diff
@@ -0,0 +1,314 @@
+#include <algorithm>
+
+#include "printer.h"
+
+namespace CSymPy {
+
+StrPrinter::StrPrinter(bool ordered) : BaseVisitor(this), order_(ordered) {
+
+}
+void StrPrinter::bvisit(const Basic &x) {
+    std::cout << "Basic" <<std::endl;
+}
```

Comment:
Can this be ever invoked? If not, it should raise an exception. If so, then it should have the original code in here:
```c++
    std::ostringstream s;
    s << "<" << typeName<Basic>(*this)
        << " instance at " << (const void*)this << ">";
    return s.str();
```